Source/core/CyclicBuffer.cpp

```diff
@@ -120,8 +146,34 @@ namespace Core {
             if (initiator == true) {
 
 #ifndef __WINDOWS__
-                _administration->_signal = PTHREAD_COND_INITIALIZER;
-                _administration->_mutex = PTHREAD_MUTEX_INITIALIZER;
+                pthread_condattr_t cond_attr;
+
+                // default values
+                int ret = pthread_condattr_init(&cond_attr);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                ret = pthread_condattr_setpshared(&cond_attr, PTHREAD_PROCESS_SHARED);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                ret = pthread_cond_init(&(_administration->_signal), &cond_attr);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                pthread_mutexattr_t mutex_attr;
+
+                // default values
+                ret = pthread_mutexattr_init(&mutex_attr);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+                // enable access for threads, also in different processes
+                ret = pthread_mutexattr_setpshared(&mutex_attr, PTHREAD_PROCESS_SHARED);
+                ASSERT(ret == 0); DEBUG_VARIABLE(ret);
+
+#ifndef __DEBUG__
```

Comment:
Reading up on the documentation, this flag triggers that a recursive lock (lock already taken by this thread) will return an error. In the context of the CyclicBuffer, I think the lock should indeed *not* be recursive so returning an error would be good. But why is this only applicable if we build this in non debug builds? Why would we not check this in Debug builds ?
And why is the functional code now in the ASSERT?

Reply:
PTHREAD_MUTEX_ERRORCHECK is more expensive, and, as such, not the preferred type. PTHREAD_MUTEX_NORMAL usually suffices and is the default. Assert's are part of the debug(ging) process and additional error checking is done in this context.